python/test/gluon/test_frontend.py

```diff
@@ -23,20 +32,65 @@ def anonymize_ir(ir):
     return TARGET_PAT.sub('ttg.target = "..."', ir)
 
 
+def get_common_test_aot_target() -> GPUTarget:
+    """Returns a common test suitable AOT target from the current runtime."""
+    if is_hip():
+        # We use gfx1200 as the target for testing's purpose given it has warp size == 32
+        # and is easier to write common expecttest checks.
+        return GPUTarget("hip", "gfx1200", 32)
+    # For others, use the current runtime's target.
+    return triton.runtime.driver.active.get_current_target()
```

Comment:
I think we should make the test fully GPU independent so that they can even run on CPU only machine.

Reply:
Yeah actually I was a bit thinking about which path to go earlier. Thought it's good to scale automatically as we add new architectures. But it's less explicit and anchors on GPU machines still. Changed to do `@pytest.mark.parametrize` as you suggested.